bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: expose supported Kafka sizes as public endpoint

## Description

Purpose of this change is to expose supported Kafka instance types through a public endpoint. 
This endpoint returns a list of supported Kafka instance types based on cloud provider and cloud region. Each returned instance type contains a list of supported sizes.
The endpoint requires two parameters:
- `cloud_provider` - Name of the supported cloud provider eg. "aws"
- `cloud_region` - Name of the supported cloud region eg. "us-east-1"
## Verification Steps

1. Run kas-fleet-manager locally with this change.
2. Run the `make run/docs` command to run Swagger. The endpoint requires use of Bearer token.
3. The endpoint for returning the list of supported Kafka instance types is listed as:
   `/api/kafkas_mgmt/v1/instance_types/{cloud_provider}/{cloud_region}`
4. The endpoint requires a supported cloud provider name and supported cloud region name. The following parameters can be used:
   cloud_provider: aws
   cloud_region:  us-east-1
5. Endpoint should returned the list of supported Kafka instance types and sizes.
   Example return:
```
{
  "instance_types": [
    {
      "id": "developer",
      "display_name: "Trial"
      "sizes": [
        {
          "id": "x1",
          "ingress_throughput_per_sec": {
            "bytes": 31457280
          },
          "egress_throughput_per_sec": {
            "bytes": 31457280
          },
          "total_max_connections": 3000,
          "max_data_retention_size": {
            "bytes": 1073741800000
          },
          "max_partitions": 1000,
          "max_data_retention_period": "P14D",
          "max_connection_attempts_per_sec": 100,
          "quota_consumed": 1,
          "quota_type": "rhosak",
          "capacity_consumed": 1
        }
      ]
    },
    {
      "id": "standard",
      "display_name: "Standard"
      "sizes": [
        {
          "id": "x1",
          "ingress_throughput_per_sec": {
            "bytes": 52428800
          },
          "egress_throughput_per_sec": {
            "bytes": 104857600
          },
          "total_max_connections": 3000,
          "max_data_retention_size": {
            "bytes": 1073741800000
          },
          "max_partitions": 1500,
          "max_data_retention_period": "P14D",
          "max_connection_attempts_per_sec": 100,
          "quota_consumed": 1,
          "quota_type": "rhosak",
          "capacity_consumed": 1
        }
      ]
    }
  ]
}
```
## Checklist (Definition of Done)
